src/native/crt.c

```diff
@@ -286,6 +286,18 @@ bool aws_jni_check_and_clear_exception(JNIEnv *env) {
     return exception_pending;
 }
 
+bool aws_jni_get_and_clear_exception(JNIEnv *env, jthrowable *out) {
+    bool exception_pending = (*env)->ExceptionCheck(env);
+    if (exception_pending) {
+        (*env)->DeleteGlobalRef(env, *out);
+        if (out != NULL) {
+            *out = (jthrowable)(*env)->NewGlobalRef(env, (*env)->ExceptionOccurred(env));
```

Comment:
Good catch. I tried to make it NULL-tolerant but missed the `(*env)->DeleteGlobalRef(env, *out);`. I have removed the `if (out != NULL)` condition, since other functions like [aws_jni_byte_array_from_cursor](https://github.com/awslabs/aws-crt-java/blob/main/src/native/crt.c#L324) are also not NULL-tolerant.